### PR TITLE
Fixes Nanotrasen paper bin + now its tgui_alert instead of alert

### LINKED ISSUE #26450: You can magically pull papers from Nanotrasen paper bins from any location.
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1641

### Issue Description

Using a nanotrasen paper bin will allow you to pull one of its papers out from thin air at any location.

### What did you expect to happen?

I expected my input to be denied due to the distance I was at from the paper bin.

### What happened instead?

When I walked away and selected a blank paper, it appeared in my hand.

### Why is this bad/What are the consequences?

It's not exactly bad, but it's definitely not supposed to happen at all.

### Steps to reproduce the issue.

Interact with a Nanotrasen paper bin within proper distance. Walk away from the bin with the selection menu still open, then select letterhead or blank from any distance and it will appear in your hand.

### When did the problem start happening?

I've only discovered it today, August 9th.

### Extra information

An admin said that "they didn't sanitize the input". Whatever that means, I hope it's helpful.

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by npukoJl**
## What Does This PR Do

1. Fixes bug 
2. Replaces old alert with a modern looking one
## Why It's Good For The Game

Fixes #26450
Replaces old alert() with tgui_alert()
## Images of changes

Nothing
## Testing

https://github.com/user-attachments/assets/da8d6a2a-3c3c-4583-81f2-c7638267dc94
<hr>

### Declaration
- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/feature_freeze/CODE_OF_CONDUCT.md#currently-changes-to-the-following-types-of-content-reuires-pre-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.

<hr>

## Changelog
:cl:
fix: Nanotrasen Paperbin now does not let you take paper from it if you are too far away
/:cl:

### CODE REVIEW

File: code/modules/paperwork/paperbin.dm [hunk 1]
```diff
@@ -73,10 +73,15 @@
...
-			else
-				P = new /obj/item/paper
+			var/choice = tgui_alert(user, "Choose a style", "Paperbin", list("Letterhead","Blank", "Cancel"))
+			if(!choice || (choice == "Cancel") || !Adjacent(user))
+				return
+			switch(choice)
+				if("Letterhead")
+					P = new letterhead_type
+				if("Blank")
+					P = new /obj/item/paper
```
Comment (AyIong) [inline code/modules/paperwork/paperbin.dm:83]:
```suggestion
			var/choice = tgui_alert(user, "Choose a style", "Paperbin", list("Letterhead","Blank", "Cancel"))
			if(isnull(choice) || !Adjacent(user))
				return
				
			switch(choice)
				if("Letterhead")
					P = new letterhead_type
					
				if("Blank")
					P = new /obj/item/paper
					
				if("Cancel")
					return
```

**Review by AyIong — CHANGES_REQUESTED** [review]
When you use `tgui_input`, you has to use `isnull(value)` instead of !value
Cause if user close window, it will something and didn't pass !value

And then, you've got a switch, it seems logical to me to put "Cancel" in there.

File: code/modules/paperwork/paperbin.dm (hunk 1 above)
Comment (MixerBaikalLover) [reply code/modules/paperwork/paperbin.dm]: Done as you suggested, TY <3

File: code/modules/paperwork/paperbin.dm [hunk 2]
```diff
@@ -73,10 +73,17 @@
 			P = papers[length(papers)]
 			papers.Remove(P)
 		else
-			if(letterhead_type && alert("Choose a style", null,"Letterhead","Blank")=="Letterhead")
-				P = new letterhead_type
-			else
-				P = new /obj/item/paper
+			var/choice = tgui_alert(user, "Choose a style", "Paperbin", list("Letterhead","Blank", "Cancel"))
```
Comment (Contrabang) [inline code/modules/paperwork/paperbin.dm:76]:
```suggestion
			var/choice = tgui_alert(user, "Choose a style", "Paperbin", list("Letterhead", "Blank", "Cancel"))
```

File: code/modules/paperwork/paperbin.dm [hunk 3]
```diff
@@ -73,10 +73,17 @@
...
+			if(isnull(choice) || !Adjacent(user))
+				return
+			switch(choice)
+				if("Letterhead")
+					P = new letterhead_type
+				if("Blank")
+					P = new /obj/item/paper
+				if("Cancel")
+					return
```
Comment (Contrabang) [inline code/modules/paperwork/paperbin.dm:85]:
```suggestion
			if(isnull(P))
				return
```

**Review by Contrabang — DISMISSED** [review]

File: code/modules/paperwork/paperbin.dm (hunk 3 above)
Comment (MixerBaikalLover) [reply code/modules/paperwork/paperbin.dm:85]: I dont get removal "Cancel" from switch-case

Comment (MixerBaikalLover) [reply code/modules/paperwork/paperbin.dm:85]: but i will add null-check on P right after switch-case just in case

Comment (Sirryan2002) [reply code/modules/paperwork/paperbin.dm:85]: this could just be `else` tbh

**Review by Sirryan2002 — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Comment by S34NW** [comment]
Downstream contributor, has been advised to link account for future PRs

**Pull request merged**